src/coreclr/nativeaot/Common/src/Internal/Runtime/CompilerHelpers/StartupCodeHelpers.cs

```diff
@@ -217,6 +232,49 @@ private static unsafe object[] InitializeStatics(IntPtr gcStaticRegionStart, int
 
             return spine;
         }
+
+        private static unsafe void RehydrateData(IntPtr dehydratedData, int length)
+        {
+            // Destination for the hydrated data is in the first 32-bit relative pointer
+            byte* pDest = (byte*)ReadRelPtr32((void*)dehydratedData);
+
+            // The dehydrated data follows
+            byte* pCurrent = (byte*)dehydratedData + sizeof(int);
+            byte* pEnd = (byte*)dehydratedData + length;
+
+            // Fixup table immediately follows the command stream
+            int* pFixups = (int*)pEnd;
+
+            while (pCurrent < pEnd)
+            {
+                pCurrent = DehydratedDataCommand.Decode(pCurrent, out int command, out int payload);
+                switch (command)
+                {
+                    case DehydratedDataCommand.Copy:
+                        // TODO: can we do any kind of memcpy here?
```

Comment:
`Buffer.MemoryCopy` ?

Reply:
We can't call that from Test.CoreLib. I'm also a little bit worried - this code runs during very early startup - casting, typeof, `newobj` and many other things are not available. Buffer.MemoryCopy feels a bit higher up the stack that I'm not sure it would be safe to call.

Reply:
Yes this runs very early. I am not sure if Buffer.MemoryCopy does anything complex, but since the data here is unmanaged, it would eventually just call something like `InternalCalls.memmove`.

Are these copied chunks long enough to involve memmove? If they are typically just 10-20 bytes, it may not get any faster.